jax/dtypes.py

```diff
@@ -141,38 +141,72 @@ def issubdtype(a, b):
 can_cast = onp.can_cast
 issubsctype = onp.issubsctype
 
-_bfloat16_type_promotions = {
-  onp.dtype('bool'): onp.dtype(bfloat16),
-  onp.dtype(bfloat16): onp.dtype(bfloat16),
-  onp.dtype('float16'): onp.dtype('float32'),
-  onp.dtype('float32'): onp.dtype('float32'),
-  onp.dtype('float64'): onp.dtype('float64'),
-  onp.dtype('complex64'): onp.dtype('complex64'),
-  onp.dtype('complex128'): onp.dtype('complex128'),
-  onp.dtype('int8'): onp.dtype(bfloat16),
-  onp.dtype('int16'): onp.dtype('float32'),
-  onp.dtype('int32'): onp.dtype('float64'),
-  onp.dtype('int64'): onp.dtype('float64'),
-  onp.dtype('uint8'): onp.dtype(bfloat16),
-  onp.dtype('uint16'): onp.dtype('float32'),
-  onp.dtype('uint32'): onp.dtype('float64'),
-  onp.dtype('uint64'): onp.dtype('float64'),
-}
+
+# List of all valid JAX dtypes, in the order they appear in the type promotion
+# table.
+_jax_types = [
+  onp.dtype('bool'),
+  onp.dtype('int8'),
+  onp.dtype('int16'),
+  onp.dtype('int32'),
+  onp.dtype('int64'),
+  onp.dtype('uint8'),
+  onp.dtype('uint16'),
+  onp.dtype('uint32'),
+  onp.dtype('uint64'),
```

Comment:
If it’s not too annoying, maybe flip the order of signed and unsigned integers? Then the dtypes kinds would be sorted by priority.

Reply:
Done.